contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -268,20 +264,23 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
             blockchainID = WARP_MESSENGER.getBlockchainID();
         }
 
+        // Parse the payload of the message.
+        TeleporterMessage memory teleporterMessage = abi.decode(
+            warpMessage.payload,
+            (TeleporterMessage)
+        );
+
         // Require that the message was intended for this blockchain and teleporter contract.
         require(
-            warpMessage.destinationChainID == blockchainID,
+            teleporterMessage.destinationChainID == blockchainID,
             "TeleporterMessenger: invalid destination chain ID"
         );
-        require(
-            warpMessage.destinationAddress == address(this),
-            "TeleporterMessenger: invalid destination address"
-        );
 
-        // Parse the payload of the message.
-        TeleporterMessage memory teleporterMessage = abi.decode(
-            warpMessage.payload,
-            (TeleporterMessage)
+        // Require the source teleporter address matches this contract.
+        // It is previously encoded as destinationAddress in WarpMessage before subnet-evm@v0.5.7
+        require(
+            teleporterMessage.sourceTeleporterAddress == address(this),
```

Comment:
do we need the `sourceTeleoprterAddress`? In the `WarpMessage` origin sender address is included, and we already that it's from the Teleporter address.

Reply:
I think this is mirroring the check against the now removed `warpMessage.destinationAddress`.

We might not need that check at all though... We already verify that the sender is the same Teleporter contract address, so the check against the intended destination seems redundant. I agree with Matt that we can remove the field `sourceTeleporterAddress` from `TeleporterMessage`

Reply:
removed